### PR TITLE
QOLDEV-817 add lambda function to run instance setup commands

### PR DESCRIPTION
**Pull request by ThrawnCA**
We need to run Systems Manager commands on instance start, but we don't want to give servers the privileges to run arbitrary Systems Manager commands, since that's an easy route to privilege escalation. Instead we can delegate the job to a Lambda function (which is much harder to subvert) with the appropriate role.

### CODE REVIEW

File: templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2 [hunk 1]
```diff
@@ -247,6 +247,74 @@ Resources:
...
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - !Ref RunCommandPolicy
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2:296]: is this being attached to a lambda or to a ec2 instance.

File: templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2 [hunk 2]
```diff
@@ -247,6 +247,74 @@ Resources:
...
+            Resource:
+              - !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
+            Condition:
+              StringEquals:
+                "ssm:resourceTag/Environment":
+                  !Ref Environment
+                "ssm:resourceTag/Service":
+                  !Ref ApplicationName
+
+  SetupInstanceRole:
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2:284]: Please align with the InstanceSetupLambda i.e. InstanceSetupLambdaRole

File: templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2 [hunk 3]
```diff
@@ -494,50 +562,6 @@ Resources:
         Type: !Ref CookbookURLType
         # SshKey: !Sub "${CookbookSSHKey}\n"
         Url: !Ref CookbookURL
-      CustomJson:
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2:497]: does opsworks need this when we are transitioning?

**Review by duttonw — APPROVED** [review]

File: templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2 (hunk 1 above)
Comment (ThrawnCA) [reply templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2:296]: It was originally written with the idea of attaching to EC2 but we decided that would be insecure and Lambda is better.

File: templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2 (hunk 3 above)
Comment (ThrawnCA) [reply templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2:497]: Not since cookbook 7.0.4. It's all in Parameter Store now.

**Pull request merged**